src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -203,6 +227,25 @@ private async Task<T> GetConfigurationNonBlockingAsync(CancellationToken cancel)
 
                 try
                 {
+                    // Check if event handler can provide configuration
+                    if (ConfigurationEventHandler != null)
```

Comment:
nit: could pull this into a property and it would read better e.g.

if (UsingEventHandler)